### ISSUE TITLE
Crash with Blit Image on a drawn screen

### ISSUE THREAD

**Issue by ldir-hector** [spotlessmind1975/ugbasic#604]
I have tried to use the brand new BLIT IMAGE with images that I have loaded from the screen itself.

Here is the code : 
```
'Blit image
BITMAP ENABLE(160,200,16)
CLS

COLOR BORDER RGB(0,0,255)
VAR boom AS IMAGE, ciel AS IMAGE, plancheSprite AS IMAGE
VAR back AS IMAGE
plancheSprite = LOAD IMAGE("explosion_sprites.png")
PUT IMAGE plancheSprite AT 0,0 WITH TRANSPARENCY
boom := NEW IMAGE(14,17)
ciel := NEW IMAGE(14,17)
back := NEW IMAGE(158,32)
GET IMAGE boom FROM 1,1
GET IMAGE ciel FROM 15,1

'PUT IMAGE boom AT 80,100
'PUT IMAGE ciel AT 100,100
PEN COLOR(11)
BAR 0,100 TO 158,132
GET IMAGE back FROM 0,100
CLS
BLIT putimage AS (SOURCE 1)
BLIT puttransmask AS (((DESTINATION) MASKED (INVERSE(THRESHOLD SOURCE 2))) OR (SOURCE 1))
				
BLIT IMAGE back AT 0,0 WITH putimage
BLIT IMAGE boom,ciel AT 0,0 WITH puttransmask
BLIT IMAGE ciel AT 0,100 WITH putimage
PRINT "Trace 1"
```
And the program crashed showing this : 
![image](https://github.com/spotlessmind1975/ugbasic/assets/118670680/67a993a7-3ca0-4c56-a87a-dd48a89bbf3f)

I only rely on the articles you made on facebook, but the documentation shall be completed with them, because facebook is not reliable for keeping documentation.

**Comment by ldir-hector** [comment]
[essaiBlit.cpc.dsk.zip](https://github.com/spotlessmind1975/ugbasic/files/11454608/essaiBlit.cpc.dsk.zip)

My DSK

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector and **thank you for your bug report!**

I'm checking the reason for the problem, and I hope to give an answer soon. As for the documentation, it will be completed before the release of the next release, which will also include another target.

Thank you again!

**Comment by spotlessmind1975** [comment]
Hi @ldir-hector, the problem was in calculating the width of the buffer to retrieve with the `GET IMAGE` command, which erroneously used the `SRA` Z80 opcode instead of `SRL`. This meant that if the size was greater than 127 pixels, it would overwrite the memory where the program was located, creating the problem.

Now everything is fixed with COLDFIX v1.13.3 [rev. 20230512.0020].

I take the liberty of suggesting the use of the direct assignment operator (`:=`) anywhere, at least when dealing with graphical and audio resources. This saves memory and avoids saturating available memory for the program, which is usually another reason things don't work.

Thank you and have a nice retrocoding!
